OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -43,16 +42,19 @@ public PickupUnit(Actor self, Actor cargo, int delay)
 			carryableFacing = cargo.Trait<IFacing>();
 			carryableBody = cargo.Trait<BodyOrientation>();
 
-			movement = self.Trait<IMove>();
 			carryall = self.Trait<Carryall>();
-			carryallFacing = self.Trait<IFacing>();
 
-			state = PickupState.Intercept;
+			ChildHasPriority = false;
 		}
 
 		protected override void OnFirstRun(Actor self)
 		{
 			carryall.ReserveCarryable(self, cargo);
```

Comment:
Replacing the contents of `OnFirstRun` with:
```
			if (carryall.ReserveCarryable(self, cargo))
			{
				// Fly to the target and wait for it to be locked for pickup
				// These activities will be cancelled and replaced by Land once the target has been locked
				QueueChild(new Fly(self, Target.FromActor(cargo)));
				QueueChild(new FlyIdle(self, tickIdle: false));
			}
```
resolves the glitch when already carrying another unit